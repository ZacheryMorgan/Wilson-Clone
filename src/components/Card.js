import React from "react";
import "../styles/Card.scss";

const Card = ({ data, onChange, cart }) => {
  return (
    <div className="card-wrapper  shadow-md shadow-black rounded-sm">
      <div className="img-wrapper relative overflow-hidden">
        <img
          className="w-72 rounded-sm hover:scale-125"
          src={data.img}
          alt=""
          loading="lazy"
        />
        <p className="name absolute bottom-0 bg-white px-3 py-1 rounded-sm -translate-x-1/2 left-1/2 font-bold">
          {data.name}
        </p>
      </div>
      <section
        className="price-wrapper
         flex justify-between items-center px-7 h-12"
      >
        <p className="price font-bold">${data.price}</p>
        <input
          className="w-14"
          type="number"
          name=""
          id=""
          placeholder="0"
          value={data.quantity}
          min="0"
          onChange={(e) => onChange(e, data.name)}
        />
      </section>
    </div>
  );
};

export default Card;

// hover:scale-110
