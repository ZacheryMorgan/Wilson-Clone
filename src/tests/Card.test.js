import React from 'react';
import { render } from '@testing-library/react';
import Card from '../components/Card';

describe("Card component", () => {
  it("renders card component correctly", () => {
    const { container } = render(<Card />)
    expect(container).toMatchSnapshot()
  })
})