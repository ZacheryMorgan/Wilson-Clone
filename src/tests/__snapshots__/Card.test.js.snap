// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders card component correctly 1`] = `
<div>
  <div
    class="card-wrapper  shadow-md shadow-black rounded-sm"
  >
    <div
      class="img-wrapper relative"
    >
      <img
        alt=""
        class="w-96"
        src="./assets/Shell1.jpg"
      />
      <p
        class="name absolute bottom-0 bg-white px-3 py-1 rounded-sm -translate-x-1/2 left-1/2 font-bold"
      >
        Kayle's Shell
      </p>
    </div>
    <section
      class="price-wrapper
 flex justify-between items-center px-7 h-12"
    >
      <p
        class="price font-bold"
      >
        $10.00
      </p>
      <input
        class="w-14"
        id=""
        name=""
        placeholder="0"
        type="number"
      />
    </section>
  </div>
</div>
`;
